### ISSUE TITLE
Problems with adding hyperlinks

### ISSUE THREAD

**Issue by gnefnuy** [xlwings/xlwings#1125]

**Comment by fzumstein** [comment]
Please add an example of how you would do it in either VBA or with cell formulas, that makes it easier to help.

**Comment by gnefnuy** [comment]
I found this
[https://docs.microsoft.com/en-us/office/vba/api/excel.hyperlinks.add](https://docs.microsoft.com/en-us/office/vba/api/excel.hyperlinks.add)
I think it should be the `SubAddress` parameter.
```vb
Option Explicit

Sub Sample()
    Dim rowNumb As Long
    Dim ws As Worksheet

    '~~> Change this to the relevant number
    rowNumb = 1

    If TypeOf Selection Is Range Then
        On Error Resume Next
        Set ws = ThisWorkbook.Sheets(rowNumb + 1)
        If Err.Number <> 0 Then
            MsgBox "Sheet doesn't exist"
            Exit Sub
        End If
        On Error GoTo 0

        ThisWorkbook.ActiveSheet.Hyperlinks.Add Anchor:=Selection, _
                                   Address:="", _
                                   SubAddress:=ws.Name & "!A1", _
                                   TextToDisplay:="Link"
    Else
        MsgBox "InValid Range Object"
    End If
End Sub
```

**Comment by fzumstein** [comment]
thanks!

**Comment by pickshell** [comment]
Jump links between different sheet pages in the same workbook, hope to have a simple and general solution！！
（Windows 10  0.18.0, Office 2016, Python 3.7)

wb = xw.Book('test.xlsx')
wb.sheets[0].api.Hyperlinks.Add(Anchor=wb.sheets[0].range((1,1)).api,Address="",SubAddress=wb.sheets[1].name+"!A1",ScreenTip="",TextToDisplay=wb.sheets[1].name)
    wb.sheets[0].api.Hyperlinks.Add(Anchor=wb.sheets[0].range((2,1)).api,Address="",SubAddress=wb.sheets[2].name+"!A1",ScreenTip="",TextToDisplay=wb.sheets[2].name)

**Comment by chuong-pivot** [comment]
> wb = xw.Book('test.xlsx')
wb.sheets[0].api.Hyperlinks.Add(Anchor=wb.sheets[0].range((1,1)).api,Address="",SubAddress=wb.sheets[1].name+"!A1",ScreenTip="",TextToDisplay=wb.sheets[1].name)
wb.sheets[0].api.Hyperlinks.Add(Anchor=wb.sheets[0].range((2,1)).api,Address="",SubAddress=wb.sheets[2].name+"!A1",ScreenTip="",TextToDisplay=wb.sheets[2].name)

Hi, are there any way for doing this on macos?
Currently as i see in docs and debugging is i will get the hyperlink on `Web Page or File` option by using [add_hyperlink](https://docs.xlwings.org/en/stable/api/range.html#xlwings.Range.add_hyperlink), but i want the `This Document` option instead.

<img width="525" alt="image" src="https://github.com/user-attachments/assets/5486a893-8995-4158-a22f-3c0a5e315975">

When debugging, i'm aware that manually setting hyperlink inside excel file would assign the address to sub_address instead of address, which `add_hyperlink` won't assign the address to.

Below image are logs of 2 hyperlinks, first 5 lines from links i create manually, last 5 lines are from `add_hyperlink` function:
```
print(link)
print(link.address())
print(link.sub_address())
print(link.screen_tip())
print(link.text_to_display())
```
![image](https://github.com/user-attachments/assets/df8924b6-ad39-40e5-99f9-f25f7d54cb08)

**Long story short, how do i add a `This Document` hyperlink option on macos? Can i achieve this with `sheet.api` or `cell.api` or any work around?**

I have tried things like `cell.api.add_hyperlink()`, `cell.api.new_hyperlink()`, `cell.api.hyperlinks.add()`, `cell.api.hyperlinks().add()` but nothing work so far :(
Even a link to the `cell.api` docs so i can see what functions, variables it's having would be very appreciate 🙏

**Comment by fzumstein** [comment]
I don't know what the syntax is off the top of my head, but have a look at: https://github.com/xlwings/xlwings/blob/main/DEVELOPER_GUIDE.md#macos

**Comment by chuong-pivot** [comment]
Thank you!! i was able to add `This Document` hyperlink after doing some changes on xlwings local package 🙏 
![image](https://github.com/user-attachments/assets/0898ec26-43d4-4b95-93a3-1795406b0bf3)
![image](https://github.com/user-attachments/assets/bdd06428-86d8-4b6d-b1f8-a69ffa3253d9)
